bundles/org.eclipse.ui.editors/src/org/eclipse/ui/internal/texteditor/stickyscroll/StickyScrollingControl.java

```diff
@@ -244,14 +242,19 @@ private void styleStickyLines() {
 			return;
 		}
 
-		List<StyleRange> stickyLinesStyleRanges= new ArrayList<>();
-		int stickyLineTextOffset= 0;
-		for (int i= 0; i < getNumberStickyLines(); i++) {
-			StickyLine stickyLine= stickyLines.get(i);
-			stickyLinesStyleRanges.addAll(getStickyLineStyleRanges(stickyLine, stickyLineTextOffset));
-			stickyLineTextOffset+= stickyLine.text().length() + System.lineSeparator().length();
+		int stickyLineOffset= 0;
+		for (IStickyLine stickyLine : stickyLines) {
+			StyleRange[] styleRanges= stickyLine.getStyleRanges();
+
+			if (styleRanges != null) {
+				for (StyleRange styleRange : styleRanges) {
+					styleRange.start+= stickyLineOffset;
+					stickyLineText.setStyleRange(styleRange);
```

Comment:
Performance question: calling setStyleRanges() and passing all styleRanges in one call might be faster than calling setStyleRange separately for each range? I have not made any measurements, it is just an idea.

Reply:
Good point, done. Thanks for the review.